assembl/views/__init__.py

```diff
@@ -362,9 +363,78 @@ def process_export_list(ls):
         providers=providers,
         providers_json=json.dumps(providers),
         translations=io.open(jedfilename, encoding='utf-8').read(),
-        admin_email=admin_email
+        admin_email=admin_email,
     )
 
+    base.update({
+        "opengraph_locale": get_opengraph_locale(request),
+        "get_description": get_description(request),
+        "get_landing_page_image": get_landing_page_image(),
+        "private_social_sharing": private_social_sharing(),
+        "get_topic": get_topic(request)
+    })
+
+    return base
+
+
+def private_social_sharing():
+
+    from ..auth.util import get_current_discussion
+    discussion = get_current_discussion()
+    if discussion:
+        return discussion.preferences["private_social_sharing"]
+    else:
+        return False
+
+
+def get_opengraph_locale(request):
+    from ..auth.util import get_user, get_current_discussion
+    from assembl.lib.locale import strip_country
+    user = get_user(request)
+    discussion = get_current_discussion()
+    if user is None and discussion is None:
+        locale = "fr"
+    elif discussion and user is None:
+        locale = discussion.preferences['preferred_locales'][0]
+    elif user:
+        locale = user.language_preference[0].locale.code
+        locale = strip_country(locale)
+    return locale
+
+
+def get_description(request):
+    """Returns the description corresponding to the locale of the current discussion"""
+    opengraph_locale = get_opengraph_locale(request)
+    from ..auth.util import get_current_discussion
+    discussion = get_current_discussion()
+    if discussion:
+        if discussion.preferences["extra_json"]["objectives"]["descriptionEntries"][opengraph_locale]:
```

Comment:
This code is not defensive enough. In the `extra_json`, if `objectives` is not set (which is very possible), then this will crash and the page will fail to load. Code more defensively. You can use python `dict`'s `get` method to pass a default value.